File: POO_Exemplos/Java/POO.java
// Abstração + Encapsulamento
public class Conta {
    private double saldo;

    public void depositar(double valor) {
        if (valor > 0) saldo += valor;
    }

    public double getSaldo() {
        return saldo;
    }
}

// Herança + Polimorfismo
class Animal {
    public void emitirSom() {
        System.out.println("Som genérico");
    }
}

class Cachorro extends Animal {
    @Override
    public void emitirSom() {
        System.out.println("Latido");
    }
}

File: POO_Exemplos/CSharp/POO.cs
// Abstração + Encapsulamento
public class Conta
{
    private double saldo;

    public void Depositar(double valor)
    {
        if (valor > 0) saldo += valor;
    }

    public double GetSaldo()
    {
        return saldo;
    }
}

// Herança + Polimorfismo
public class Animal
{
    public virtual void EmitirSom()
    {
        Console.WriteLine("Som genérico");
    }
}

public class Cachorro : Animal
{
    public override void EmitirSom()
    {
        Console.WriteLine("Latido");
    }
}